client/search/search-input.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import SearchIcon from '../icons/material/baseline-search-24px.svg'
+import { useKeyListener } from '../keyboard/key-listener'
+import { fastOutSlowInShort } from '../material/curves'
+import TextField from '../material/text-field'
+import { useValueAsRef } from '../state-hooks'
+
+const ESCAPE = 'Escape'
+const F = 'KeyF'
+
+const TextFieldContainer = styled(TextField)`
+  width: ${props => (props.isFocused ? '250px' : '200px')};
```

Comment:
this is a weird value to bake into a component that you seemingly intend to be generic across a bunch of interfaces. Would prefer that this is a percentage of some sort, or that it is at least customizable through CSS custom properties